app/models/bundle_instance.rb

```diff
@@ -0,0 +1,12 @@
+class BundleInstance < ApplicationRecord
+  belongs_to :bundle
+  belongs_to :site
+  has_many :application_instances
+  has_many :applications, through: :bundle
+
+  def create_application_instances
+    bundle&.applications&.map do |app|
```

Comment:
Is there a reason to prefer each over map? I generally prefer map over each because I think it leads to less refactoring.